app/assets/javascripts/discourse/app/instance-initializers/video-placeholder.js

```diff
@@ -0,0 +1,44 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import getURL from "discourse-common/lib/get-url";
+
+export default {
+  initialize() {
+    withPluginApi("0.8.7", (api) => {
+      function _handleEvent(event) {
+        const img = event.target;
+        const parentDiv = img.parentElement;
+
+        img.style.display = "none";
+
+        const videoHTML = `
+        <video width="100%" height="100%" preload="metadata" controls>
+          <source src="${parentDiv.dataset.videoSrc}" ${parentDiv.dataset.origSrc}>
+          <a href="${parentDiv.dataset.videoSrc}">${parentDiv.dataset.videoSrc}</a>
+        </video>`;
+        parentDiv.insertAdjacentHTML("beforeend", videoHTML);
+        parentDiv.classList.add("video-container");
+
+        const video = parentDiv.querySelector("video");
+        video.play();
+      }
+
+      function _attachCommands(post, helper) {
+        if (!helper) {
+          return;
+        }
+
+        let images = post.querySelectorAll(".video-placeholder-container img");
+
+        images.forEach((img) => {
+          img.src = getURL("/images/video-placeholder.svg");
```

Comment:
Oh, since the image here is essentially just the "play" svg icon, can we use `iconHTML("play")` instead? That would render an svg element, we'd need to add the proper styling for it in the stylesheets to match the design. But that icon is already included in the bundle (i.e. now new assets introduced) and it would allow themes to do some of their own styling as well.

Reply:
Ya, I like that idea.

In the future if we actually have a thumbnail image to display instead of just a blank image we would more easily be able to swap that out.

Also it would be cool to do the subtle icon wobble like we do for YouTube videos when you hover over it.

Reply:
> Also it would be cool to do the subtle icon wobble like we do for YouTube videos when you hover over it.

Yeah, that would be a nice touch, likely a tiny bit easier with an SVG.